src/components/SearchOverlay.vue

```diff
@@ -117,16 +121,46 @@ async function search() {
     }
     const variationName = breakpointService.active.value.toUpperCase();
 
+    // We do not want to render more category filter options if more than two options are already selected. (3 when allowThirdLevelCategories is enabled)
+    const selectedCategoryFilterIds = filters.value['categoryFilter'];
+    if (Array.isArray(selectedCategoryFilterIds)) {
+        renderCategoryFilterOptions.value = selectedCategoryFilterIds.length < (contextStore.context.value.allowThirdLevelCategories ? 3 : 2);
+    } else {
+        renderCategoryFilterOptions.value = true;
+    }
+    
     const request = new SearchCollectionBuilder()
         .addRequest(new ProductSearchBuilder(contextStore.defaultSettings)
             .setSelectedProductProperties(contextStore.selectedProductProperties)
             .setSelectedVariantProperties({ allData: true })
             .setTerm(filters.value.term.length > 0 ? filters.value.term : null)
-            .facets(f => f
-                .addCategoryFacet('ImmediateParent', Array.isArray(filters.value['category']) && filters.value['category']?.length > 0 ? filters.value['category'] : null)
-                .addBrandFacet(Array.isArray(filters.value['brand']) && filters.value['brand']?.length > 0 ? filters.value['brand'] : null)
-                .addSalesPriceRangeFacet('Product', applySalesPriceFacet ? Number(filters.value.price[0]) : undefined, applySalesPriceFacet ? Number(filters.value.price[1]) : undefined),
-            )
+            .filters(f => {
+                if (Array.isArray(selectedCategoryFilterIds) && selectedCategoryFilterIds.length > 0) {
+                    f.addProductCategoryIdFilter('Ancestor', selectedCategoryFilterIds[selectedCategoryFilterIds.length-1]);
+                }
+            })
+            .facets(f => {
+                f.addProductCategoryHierarchyFacet('Ancestors', undefined, { displayName: true });
+                f.addCategoryFacet('ImmediateParent', 
+                    Array.isArray(filters.value['category']) 
+                        && filters.value['category']?.length > 0 
+                        ? filters.value['category'] 
+                        : null);
+
+                f.addBrandFacet(
+                    Array.isArray(filters.value['brand'])
+                    && filters.value['brand']?.length > 0
+                        ? filters.value['brand'] 
+                        : null);
+
+                f.addSalesPriceRangeFacet('Product', 
+                    applySalesPriceFacet 
+                        ? Number(filters.value.price[0]) 
+                        : undefined,
+                    applySalesPriceFacet 
+                        ? Number(filters.value.price[1]) 
+                        : undefined);
```

Comment:
Alternativt forslag.
jeg synes dette er nemmere at læse:
```suggestion
                f.addSalesPriceRangeFacet(
                    'Product', 
                    applySalesPriceFacet ? Number(filters.value.price[0]) : undefined,
                    applySalesPriceFacet ? Number(filters.value.price[1]) : undefined);
```

Reply:
check